ocpmodels/common/utils.py

```diff
@@ -1121,3 +1136,167 @@ def scatter_det(*args, **kwargs):
+                "energy": {
+                    "fn": config["optim"].get("loss_energy", "mae"),
+                    "coefficient": config["optim"].get(
+                        "energy_coefficient", 1
+                    ),
+                },
+            }
+        ]
+        ### Define evaluation metrics
+        _eval_metrics = {
+            "metrics": {"energy": ["mae", "mse", "energy_within_threshold"]},
+        }
+        if "primary_metric" in config["task"]:
+            _eval_metrics["primary_metric"] = config["task"]["primary_metric"]
+        ### Define outputs
+        _outputs = {"energy": {"shape": 1, "level": "system"}}
+    if name == "s2ef":
+        ### Define loss functions
+        _loss_fns = [
+            {
+                "energy": {
+                    "fn": config["optim"].get("loss_energy", "mae"),
+                    "coefficient": config["optim"].get(
+                        "energy_coefficient", 1
+                    ),
+                },
+                "forces": {
+                    "fn": config["optim"].get("loss_forces", "l2mae"),
+                    "coefficient": config["optim"].get(
+                        "force_coefficient", 30
+                    ),
+                },
+            }
+        ]
+        ### Define evaluation metrics
+        _eval_metrics = {
+            "metrics": {
+                "misc": ["energy_forces_within_threshold"],
+                "energy": ["mae"],
+                "forces": [
+                    "forcesx_mae",
+                    "forcesy_mae",
+                    "forcesz_mae",
+                    "mae",
+                    "cosine_similarity",
+                    "magnitude_error",
+                ],
+            },
+        }
+        if "primary_metric" in config["task"]:
+            _eval_metrics["primary_metric"] = config["task"]["primary_metric"]
+        ### Define outputs
+        _outputs = {
+            "energy": {"shape": 1, "level": "system"},
+            "forces": {
+                "shape": 3,
+                "level": "atom",
+                "train_on_free_atoms": (
+                    config["task"].get("train_on_free_atoms", False)
+                ),
+                "eval_on_free_atoms": (
+                    config["task"].get("eval_on_free_atoms", True)
+                ),
+            },
+        }
+
+    if config["dataset"].get("normalize_labels", False):
+        normalizer = {
+            "energy": {
+                "mean": config["dataset"]["target_mean"],
+                "stdev": config["dataset"]["target_std"],
+            },
+            "forces": {
+                "mean": config["dataset"]["grad_target_mean"],
+                "stdev": config["dataset"]["grad_target_std"],
+            },
+        }
+        config["dataset"]["normalizer"] = normalizer
+
+    config["dataset"]["key_mapping"] = {"y": "energy", "force": "forces"}
+    ### Update config
+    config.update({"loss_fns": _loss_fns})
```

Comment:
This updates the config in-place, and returns. Could be scary if people don't expect this!

Maybe "update_old_config" might make more sense then don't return it?